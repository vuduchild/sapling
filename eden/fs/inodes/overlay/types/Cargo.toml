# @generated by autocargo from //eden/fs/inodes/overlay:serialization-rust-types

[package]
name = "serialization_types"
version = "0.1.0"
authors = ["Facebook Source Control Team <sourcecontrol-dev@fb.com>"]
edition = "2021"
license = "GPLv2+"
build = "thrift_build.rs"

[lib]
path = "thrift_lib.rs"
test = false
doctest = false

[dependencies]
anyhow = "1.0.75"
codegen_includer_proc_macro = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }
fbthrift = { version = "0.0.1+unstable", git = "https://github.com/facebook/fbthrift.git", branch = "main" }
futures = { version = "0.3.28", features = ["async-await", "compat"] }
once_cell = "1.12"
ref-cast = "1.0.18"
serde = { version = "1.0.185", features = ["derive", "rc"] }
serde_derive = "1.0.185"
thiserror = "1.0.49"

[build-dependencies]
thrift_compiler = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }

[features]
default = ["thrift_library_unittests_disabled"]
thrift_library_unittests_disabled = []
