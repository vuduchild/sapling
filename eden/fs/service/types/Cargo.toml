# @generated by autocargo from //eden/fs/service:thrift-rust-types

[package]
name = "thrift_types"
version = "0.1.0"
authors = ["Facebook Source Control Team <sourcecontrol-dev@fb.com>"]
edition = "2021"
license = "GPLv2+"
build = "thrift_build.rs"

[lib]
path = "thrift_lib.rs"
test = false
doctest = false

[dependencies]
anyhow = "1.0.75"
codegen_includer_proc_macro = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }
config_thrift = { version = "0.1.0", path = "../../config" }
fb303_core = { version = "0.0.0", git = "https://github.com/facebook/fb303.git", branch = "main" }
fbthrift = { version = "0.0.1+unstable", git = "https://github.com/facebook/fbthrift.git", branch = "main" }
futures = { version = "0.3.28", features = ["async-await", "compat"] }
once_cell = "1.12"
ref-cast = "1.0.18"
serde = { version = "1.0.185", features = ["derive", "rc"] }
serde_derive = "1.0.185"
sorted_vector_map = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }
thiserror = "1.0.49"

[build-dependencies]
thrift_compiler = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }

[features]
default = ["thrift_library_unittests_disabled"]
thrift_library_unittests_disabled = []
