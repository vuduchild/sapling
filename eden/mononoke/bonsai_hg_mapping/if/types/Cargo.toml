# @generated by autocargo from //eden/mononoke/bonsai_hg_mapping/if:bonsai_hg_mapping_entry_thrift-rust-types

[package]
name = "bonsai_hg_mapping_entry_thrift_types"
version = "0.1.0"
authors = ["Facebook"]
edition = "2021"
license = "GPLv2+"
build = "thrift_build.rs"

[lib]
path = "thrift_lib.rs"
test = false
doctest = false

[dependencies]
anyhow = "1.0.75"
codegen_includer_proc_macro = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }
fbthrift = { version = "0.0.1+unstable", git = "https://github.com/facebook/fbthrift.git", branch = "main" }
futures = { version = "0.3.28", features = ["async-await", "compat"] }
mercurial_thrift = { version = "0.1.0", path = "../../../mercurial/types/if" }
mononoke_types_thrift = { version = "0.1.0", path = "../../../mononoke_types/if" }
once_cell = "1.12"
ref-cast = "1.0.18"
serde = { version = "1.0.185", features = ["derive", "rc"] }
serde_derive = "1.0.185"
smallvec = { version = "1.6.1", features = ["serde", "specialization", "union"] }
thiserror = "1.0.49"

[build-dependencies]
thrift_compiler = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }

[features]
default = ["thrift_library_unittests_disabled"]
thrift_library_unittests_disabled = []
