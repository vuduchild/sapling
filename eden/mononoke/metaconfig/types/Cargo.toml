# @generated by autocargo from //eden/mononoke/metaconfig:metaconfig_types

[package]
name = "metaconfig_types"
version = "0.1.0"
authors = ["Facebook"]
edition = "2021"
license = "GPLv2+"

[dependencies]
anyhow = "1.0.75"
ascii = "1.0"
bookmarks_types = { version = "0.1.0", path = "../../bookmarks/bookmarks_types" }
clap = { version = "4.4.7", features = ["derive", "env", "string", "unicode", "wrap_help"] }
derive_more = "0.99.17"
facet = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }
mononoke_types = { version = "0.1.0", path = "../../mononoke_types" }
mysql_common = { version = "0.29.0", features = ["chrono", "default"] }
regex = "1.9.2"
rusoto_core = "0.46.0"
scuba = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }
serde = { version = "1.0.185", features = ["derive", "rc"] }
serde_derive = "1.0.185"
sql = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }
