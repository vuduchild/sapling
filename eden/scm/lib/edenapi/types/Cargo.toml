# @generated by autocargo from //eden/scm/lib/edenapi/types:[edenapi_types,macro_test]

[package]
name = "edenapi_types"
version = "0.1.0"
edition = "2021"

[[test]]
name = "macro_test"
path = "test/basic_test.rs"

[dependencies]
anyhow = "1.0.75"
blake2 = "0.10"
blake3 = { version = "=1.5.0", features = ["traits-preview"] }
bytes = { version = "1.1", features = ["serde"] }
dag-types = { version = "0.1.0", path = "../../dag/dag-types", features = ["for-tests"] }
paste = "1.0.14"
quickcheck = "1.0"
quickcheck_arbitrary_derive = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }
revisionstore_types = { version = "0.1.0", path = "../../revisionstore/types" }
serde = { version = "1.0.185", features = ["derive", "rc"] }
serde_bytes = "0.11"
serde_derive = "1.0.185"
sha1 = "0.10.5"
sha2 = "0.10.6"
thiserror = "1.0.49"
type_macros = { version = "0.1.0", path = "proc_macros" }
types = { version = "0.1.0", path = "../../types" }

[dev-dependencies]
insta_ext = { version = "0.1.0", path = "../../insta_ext" }
quickcheck_macros = "1.0"
serde_json = { version = "1.0.100", features = ["float_roundtrip", "unbounded_depth"] }

[features]
default = ["for-tests"]
for-tests = []
