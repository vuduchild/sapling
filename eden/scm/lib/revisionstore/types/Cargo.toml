# @generated by autocargo from //eden/scm/lib/revisionstore:revisionstore_types

[package]
name = "revisionstore_types"
version = "0.1.0"
authors = ["Facebook Source Control Team <sourcecontrol-dev@fb.com>"]
edition = "2021"

[lib]
path = "lib.rs"

[dependencies]
anyhow = "1.0.75"
byteorder = "1.3"
quickcheck = "1.0"
quickcheck_arbitrary_derive = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }
serde = { version = "1.0.185", features = ["derive", "rc"] }
serde_derive = "1.0.185"

[features]
default = ["for-tests"]
for-tests = []
