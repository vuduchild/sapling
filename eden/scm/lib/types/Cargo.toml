# @generated by autocargo from //eden/scm/lib/types:types

[package]
name = "types"
version = "0.1.0"
edition = "2021"

[dependencies]
anyhow = "1.0.75"
byteorder = "1.3"
lazy_static = { version = "1.4", optional = true }
quickcheck = { version = "1.0", optional = true }
quickcheck_arbitrary_derive = { version = "0.1.0", git = "https://github.com/facebookexperimental/rust-shed.git", branch = "main" }
rand = { version = "0.8", features = ["small_rng"], optional = true }
ref-cast = "1.0.18"
serde = { version = "1.0.185", features = ["derive", "rc"] }
serde_bytes = "0.11"
serde_derive = "1.0.185"
sha1 = "0.10.5"
thiserror = "1.0.49"
util = { version = "0.1.0", path = "../util" }
vlqencoding = { version = "0.1.0", path = "../vlqencoding" }

[dev-dependencies]
lazy_static = "1.4"
quickcheck = "1.0"
rand = { version = "0.8", features = ["small_rng"] }
serde_cbor = "0.11"
serde_json = { version = "1.0.100", features = ["float_roundtrip", "unbounded_depth"] }

[features]
default = ["for-tests"]
for-tests = ["lazy_static", "quickcheck", "rand"]
