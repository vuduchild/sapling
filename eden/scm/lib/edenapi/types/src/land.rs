/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This software may be used and distributed according to the terms of the
 * GNU General Public License version 2.
 */

use std::collections::HashMap;

#[cfg(any(test, feature = "for-tests"))]
use quickcheck_arbitrary_derive::Arbitrary;
use serde_derive::Deserialize;
use serde_derive::Serialize;
use type_macros::auto_wire;
use types::HgId;

use crate::ServerError;

#[auto_wire]
#[derive(Clone, Default, Debug, Deserialize, Serialize, Eq, PartialEq)]
#[cfg_attr(any(test, feature = "for-tests"), derive(Arbitrary))]
pub struct PushVar {
    #[id(0)]
    pub key: String,

    #[id(1)]
    pub value: String,
}

#[auto_wire]
#[derive(Clone, Default, Debug, Deserialize, Serialize, Eq, PartialEq)]
#[cfg_attr(any(test, feature = "for-tests"), derive(Arbitrary))]
pub struct LandStackRequest {
    #[id(0)]
    pub bookmark: String,

    #[id(1)]
    pub head: HgId,

    #[id(2)]
    pub base: HgId,

    #[id(4)]
    pub pushvars: Vec<PushVar>,
}

#[auto_wire]
#[derive(Clone, Default, Debug, Deserialize, Serialize, Eq, PartialEq)]
#[cfg_attr(any(test, feature = "for-tests"), derive(Arbitrary))]
pub struct LandStackData {
    #[id(0)]
    pub new_head: HgId,

    #[id(1)]
    pub old_to_new_hgids: HashMap<HgId, HgId>,
}

#[auto_wire]
#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
#[cfg_attr(any(test, feature = "for-tests"), derive(Arbitrary))]
pub struct LandStackResponse {
    #[id(0)]
    #[no_default]
    pub data: Result<LandStackData, ServerError>,
}
