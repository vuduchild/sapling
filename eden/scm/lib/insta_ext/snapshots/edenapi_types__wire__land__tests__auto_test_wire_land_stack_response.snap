---
source: land.rs
expression: "WireLandStackResponse::arbitrary(&mut g)"
---
{"0":{"Err":{"1":"Â€","2":2786141136387072907}}}
